server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreFormat.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.blobstore;
+
+import org.apache.lucene.store.IndexInput;
+import org.opensearch.common.blobstore.AsyncMultiStreamBlobContainer;
+import org.opensearch.common.blobstore.BlobContainer;
+import org.opensearch.common.blobstore.stream.write.WritePriority;
+import org.opensearch.common.blobstore.transfer.RemoteTransferContainer;
+import org.opensearch.common.blobstore.transfer.stream.OffsetRangeIndexInputStream;
+import org.opensearch.common.lucene.store.ByteArrayIndexInput;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Locale;
+
+/**
+ * Standard format that has only content for writes. Read interface does not exist as it not yet required. This format
+ * should be used for writing data from in-memory to remote store where there is no need for checksum and the client
+ * library for the remote store has inbuilt checksum capabilities while upload and download both. This format would
+ * serialise the data in Json format and store it on remote store as is. This does not support compression yet (this
+ * can be changed as required). In comparison to {@link ChecksumBlobStoreFormat}, this format does not add any additional
+ * metadata (like header and footer) to the content. Hence, this format does not depend on {@code CodecUtil} from
+ * Lucene library.
+ *
+ * @opensearch.internal
+ */
+public class BlobStoreFormat<T extends ToXContent> {
+
+    private final String blobNameFormat;
+
+    /**
+     * @param blobNameFormat format of the blobname in {@link String#format} format
+     */
+    public BlobStoreFormat(String blobNameFormat) {
+        this.blobNameFormat = blobNameFormat;
+    }
+
+    private String blobName(String name) {
+        return String.format(Locale.ROOT, blobNameFormat, name);
+    }
+
+    private BytesReference serialize(final T obj) throws IOException {
+        try (XContentBuilder xContentBuilder = XContentFactory.jsonBuilder()) {
+            xContentBuilder.startObject();
+            obj.toXContent(xContentBuilder, ToXContent.EMPTY_PARAMS);
+            xContentBuilder.endObject();
+            return BytesReference.bytes(xContentBuilder);
+        }
+    }
+
+    public void writeAsyncWithUrgentPriority(T obj, BlobContainer blobContainer, String name, ActionListener<Void> listener)
+        throws IOException {
+        if (blobContainer instanceof AsyncMultiStreamBlobContainer == false) {
+            write(obj, blobContainer, name);
```

Comment:
What happens if write fails? We should invoke `listener.onFailure`, right?

Reply:
If the write fails, we will get IOException. This is handled later in the RemoteIndexPathUploader.java class. We are catching the exception and invoking the onFailure of the action listener. This is applicable for the async multistream blob container as well.